src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,171 @@
+
+            // Get method containing the symbol that is being diagnosed. Should always be in a method
+            IOperation? containingMethodBody = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            if (containingMethodBody is null)
+            {
+                // This is due to VB methods having a different structure than CS methods
+                containingMethodBody = arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block);
+            }
+
+            // Create the new member
+            SyntaxNode newMember = generator.FieldDeclaration(
+                newMemberName,
+                generator.TypeExpression(arrayArgument.Type),
+                GetAccessibility(model.GetEnclosingSymbol(containingMethodBody!.Syntax.SpanStart, cancellationToken)),
+                DeclarationModifiers.Static | DeclarationModifiers.ReadOnly,
+                arrayArgument.Syntax
+            ).FormatForExtraction(containingMethodBody.Syntax);
+
+            // Add the new extracted member
+            ISymbol lastFieldSymbol = typeFields.LastOrDefault();
+            if (lastFieldSymbol is not null)
+            {
+                // Insert after last field if any fields are present
+                SyntaxNode lastFieldNode = await lastFieldSymbol.DeclaringSyntaxReferences.First().GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                editor.InsertAfter(lastFieldNode, newMember);
+            }
+            else
+            {
+                // Insert before first method if no fields are present, as a method already exists
+                editor.InsertBefore(containingMethodBody.Syntax, newMember);
+            }
+
+            // Replace argument with a reference to our new member
+            SyntaxNode identifier = generator.IdentifierName(newMemberName);
+            if (isInvoked)
+            {
+                editor.ReplaceNode(node, generator.WithExpression(identifier, node));
+            }
+            else
+            {
+                editor.ReplaceNode(node, generator.Argument(identifier));
+            }
+
+            // Return changed document
+            return await Formatter.FormatAsync(editor.GetChangedDocument(), cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
+        private static IArrayCreationOperation GetArrayCreationOperation(SyntaxNode node, SemanticModel model, CancellationToken cancellationToken,
+                out bool isInvoked)
+        {
+            // If this is a LINQ invocation, the node is already an IArrayCreationOperation
+            if (model.GetOperation(node, cancellationToken) is IArrayCreationOperation arrayCreation)
+            {
+                isInvoked = true;
+                return arrayCreation;
+            }
+            isInvoked = false;
+            return (IArrayCreationOperation)model.GetOperation(node.ChildNodes().First(), cancellationToken);
```

Comment:
What guarantees a safe cast here?

Reply:
The analyzer only creates a diagnostic for two scenarios. Those are the two we see in this method